### DISCUSSION TITLE
Getting error as "Unable to find the bootable image" when try to flash the Sample application (binary format) with Zephyr

### DISCUSSION THREAD

**Discussion by vishnusudhan** [zephyrproject-rtos/zephyr#45905] [General]
Hi,

I try to build the sample application ( Blinky /hello-world) with zephyr for i.MX RT1160 platform with MCUBoot . I can able to build & able to generate both binary & hex format image.

If i try to flash the hex format image , I can able to flash & can able to get the expected output either LED blinking or printing hello world on terminal.

If I try to flash the binary format image, I can able to flash the binary image , but I could not able to get the expected output.
If I debugged through the terminal , showing ERROR as "Unable to find the bootable image" as below,

![image](https://user-images.githubusercontent.com/104553024/169823037-2b410d23-7ef0-4266-8703-4f79d2c56c24.png)

Is there any we missed out while flashing the binary image format/ is there any restriction like can flash only hex format of application with MCUBoot?

Note: But for MCUBoot , both hex format & binary format image can be flash & able to identify the MCUBoot image.

**Comment by DerekSnell** [comment]
Hi @vishnusudhan ,

One important difference between a hex and binary file is that the hex file encodes the address for the bytes in each line of the file.  Whereas a binary has no address information for the image, and the programming tool must be told what starting address to program the binary to.  I suspect the address the runner is using for the binary is not correct, and the image is not getting flashed to the proper location in slot0 for MCUboot.

What runner are you using?  Can you confirm what address is programmed when using the binary?  And compare that to the address used by the hex file?

Best regards

**Comment by vishnusudhan** [comment]
HI @DerekSnell 

I used the command "west flash --bin-file build/zephyr/zephyr.signed.bin" to flash the sample application. through jlink segger , i am trying to flash the image

**Comment by vishnusudhan** [comment]
@DerekSnell  Is there any possibility to give the offset value using west flash command to load the binary file using jlink?

**Reply by DerekSnell** [reply]
If there is a west CLI option for the binary starting address, it is probably specific to the runner.  The general west flash help does not list an option for that:

`>west flash --help
usage: west flash [-h] [-d DIR] [-r RUNNER] [--skip-rebuild] [-H] [--board-dir DIR] [--elf-file FILE]
                  [--hex-file FILE] [--bin-file FILE] [--gdb GDB] [--openocd OPENOCD] [--openocd-search DIR]`

`Permanently reprogram a board's flash with a new binary.`

`optional arguments:
  -h, --help            show this help message and exit`

`general options:
  If the build directory is not given, the default is build/ unless the
  build.dir-fmt configuration variable is set. The current directory is
  checked after that. If either is a Zephyr build directory, it is used.`

`  -d DIR, --build-dir DIR
                        application build directory
  -r RUNNER, --runner RUNNER
                        override default runner from --build-dir
  --skip-rebuild        do not refresh cmake dependencies first`

`runner configuration:     IMPORTANT:     Individual runners support additional options not printed here.`

`  Run "west flash --context" for runner-specific options.`

`  If a build directory is found, --context also prints per-runner
  settings found in that build directory's runners.yaml file.`

`  Use "west flash --context -r RUNNER" to limit output to a
  specific RUNNER.`

`  Some runner settings also can be overridden with options like
  --hex-file. However, this depends on the runner: not all runners
  respect --elf-file / --hex-file / --bin-file, nor use gdb or openocd,
  etc.`

`  -H, --context         print runner- and build-specific help`
`  --board-dir DIR       board directory`
`  --elf-file FILE       path to zephyr.elf`
`  --hex-file FILE       path to zephyr.hex`
`  --bin-file FILE       path to zephyr.bin`
`  --gdb GDB             path to GDB`
`  --openocd OPENOCD     path to openocd`
`  --openocd-search DIR  path to add to openocd search path, if applicable`

And looking at the options for runner jlink, I don't see it there either:
`>west flash -r jlink --context`
`WARNING: no --build-dir given or found; output will be limited`
`jlink capabilities:`
`  RunnerCaps(commands={'debugserver', 'debug', 'flash', 'attach'}, dev_id=True, flash_addr=True, erase=True)`
`jlink options:`
`  -i DEV_ID, --dev-id DEV_ID`
`                        Device identifier. Use it to select the J-Link Serial Number of the device connected over USB.`
`  --dt-flash {Y,y,N,n,yes,no,YES,NO}`
`                        If 'yes', try to use flash address information from devicetree when flash addresses are`
`                        unknown (e.g. when flashing a .bin)`
`  --erase, --no-erase   mass erase flash before loading, or don't`
`  --device DEVICE       device name`
`  --id DEV_ID           Deprecated: use -i/--dev-id instead`
`  --iface IFACE         interface to use, default is swd`
`  --speed SPEED         interface speed, default is autodetect`
`  --tui                 if given, GDB uses -tui`
`  --gdbserver GDBSERVER`
`                        GDB server, default is JLinkGDBServer`
`  --gdb-host GDB_HOST   custom gdb host, defaults to the empty string and runs a gdb server`
`  --gdb-port GDB_PORT   pyocd gdb port, defaults to 2331`
`  --tool-opt TOOL_OPT   Additional options for JLink Commander, e.g. '-autoconnect 1'`
`  --commander COMMANDER`
`                        J-Link Commander, default is JLink.exe`
`  --reset-after-load, --no-reset-after-load`
`                        reset after loading? (default: no)`
